File: Samples/ContentCards/BrazeContentCardsSampleApp/BrazeContentCardsSampleApp/ViewController.m
#import "ViewController.h"
#import "AppboyContentCards.h"
#import "CustomContentCardsTableViewController.h"

@interface ViewController () <ABKContentCardsTableViewControllerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  NSString *userId = [Appboy sharedInstance].user.userID;
  if (userId != nil) {
    self.userTextField.text = userId;
  }
}

- (IBAction)displayModalContentCards:(id)sender {
  ABKContentCardsViewController *contentCards = [[ABKContentCardsViewController alloc] init];
  contentCards.contentCardsViewController.delegate = self;
  contentCards.contentCardsViewController.disableUnreadIndicator = !self.unreadIndicatorSwitch.on;
  contentCards.contentCardsViewController.navigationItem.title = @"Modal Cards";
  [self presentViewController:contentCards animated:YES completion:nil];
}

- (IBAction)displayNavigationContentCards:(id)sender {
  ABKContentCardsTableViewController *contentCards = [[ABKContentCardsTableViewController alloc] init];
  contentCards.delegate = self;
  contentCards.disableUnreadIndicator = !self.unreadIndicatorSwitch.on;
  contentCards.navigationItem.title = @"Navigation Cards";
  [self.navigationController pushViewController:contentCards animated:YES];
}

- (IBAction)displayCustomContentCards:(id)sender {
  CustomContentCardsTableViewController *contentCards = [[CustomContentCardsTableViewController alloc] init];
  contentCards.delegate = self;
  contentCards.disableUnreadIndicator = !self.unreadIndicatorSwitch.on;
  contentCards.navigationItem.title = @"Custom Cards";
  [self.navigationController pushViewController:contentCards animated:YES];
}

- (IBAction)changeUser:(id)sender {
  if (self.userTextField.text.length > 0) {
    NSString *userID = self.userTextField.text;
    [[Appboy sharedInstance] changeUser:userID];
  }
}

#pragma mark - ABKContentCardsTableViewControllerDelegate

- (BOOL)contentCardTableViewController:(ABKContentCardsTableViewController *)viewController
                 shouldHandleCardClick:(NSURL *)url {
  NSLog(@"%@", NSStringFromSelector(_cmd));
  return YES;
}

- (void)contentCardTableViewController:(ABKContentCardsTableViewController *)viewController
                    didHandleCardClick:(NSURL *)url {
  NSLog(@"%@", NSStringFromSelector(_cmd));
}

@end

File: HelloSwift/HelloSwift/ViewController.swift
import UIKit
import Appboy_iOS_SDK

class ViewController: UIViewController {
  @IBOutlet weak var userIdTextField: UITextField!
  @IBOutlet weak var customEventNameTextField: UITextField!
  @IBOutlet weak var unreadContentCardLabel: UILabel!
  @IBOutlet weak var modalOrNavigationPicker: UISegmentedControl!

  override func viewDidLoad() {
    super.viewDidLoad()
    let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
    tapGesture.numberOfTapsRequired = 1
    self.view.addGestureRecognizer(tapGesture)
    self.setUnreadContentCardLabelText()
    NotificationCenter.default.addObserver(self, selector: #selector(contentCardsUpdated(_:)), name: NSNotification.Name.ABKContentCardsProcessed, object: nil)
  }
  
  override func viewDidAppear(_ animated: Bool) {
    // Display the Braze user ID in the text field
    userIdTextField.text = Appboy.sharedInstance()?.user.userID
  }
  
  @IBAction func logCustomEventButtonTapped(sender: AnyObject) {
    let customEventName:String! = customEventNameTextField.text
    if (customEventName.count > 0) {
      Appboy.sharedInstance()?.logCustomEvent(customEventName!)
    }
  }
  
  @IBAction func updateUserId(sender: AnyObject) {
    userIdTextField.resignFirstResponder()
    Appboy.sharedInstance()?.changeUser(userIdTextField.text!)
  }

  @IBAction func contentCardButtonTapped(_ sender: Any) {
    if modalOrNavigationPicker.selectedSegmentIndex == 0 {
      let contentCards = ABKContentCardsViewController();
      contentCards.contentCardsViewController.navigationItem.title = "Modal Cards";
      self.navigationController?.present(contentCards, animated: true, completion: nil)
    } else {
      let contentCards = ABKContentCardsTableViewController();
      contentCards.navigationItem.title = "Navigation Cards"
      self.navigationController?.pushViewController(contentCards, animated: true)
    }
  }

  @objc func contentCardsUpdated(_ notification: NSNotification) {
    self.setUnreadContentCardLabelText();
  }

  func setUnreadContentCardLabelText() {
    let unreadContentCardCount = Appboy.sharedInstance()?.contentCardsController.unviewedContentCardCount() ?? 0
    let totalCardCount = Appboy.sharedInstance()?.contentCardsController.contentCardCount() ?? 0
    self.unreadContentCardLabel.text = "Unread Content Cards: \(unreadContentCardCount)/\(totalCardCount)";
    self.view.setNeedsDisplay();
  }

  @objc func dismissKeyboard() {
    self.view.endEditing(true)
  }
}
